test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -1241,6 +1241,38 @@ public void CreateValidationMetadata_NoRequiredAttribute_IsRequiredLeftAlone(boo
             Assert.Equal(initialValue, context.ValidationMetadata.IsRequired);
         }
 
+        [Fact]
+        public void CreateValidationMetadata_WillAddValidationAttributes_From_CompositeValidationAttribute()
```

Comment:
These tests look good. However we should add a few functional tests that verify this E2E (including verifying if client side validation) is correctly inferred correctly.

Reply:
Thank you for your feedback! Can I add the tests in InputObjectValidationTests.cs or should I create new test class for this?